perses/annihilation/new_relative.py

```diff
@@ -0,0 +1,1213 @@
+        """
+        Initialize the Hybrid topology factory.
+
+        Parameters
+        ----------
+        topology_proposal : perses.rjmc.topology_proposal.TopologyProposal object
+            TopologyProposal object rendered by the ProposalEngine
+        current_positions : [n,3] np.ndarray of float
+            The positions of the "old system"
+        new_positions : [m,3] np.ndarray of float
+            The positions of the "new system"
+        """
+        self._topology_proposal = topology_proposal
+        self._old_system = copy.deepcopy(topology_proposal.old_system)
+        self._new_system = copy.deepcopy(topology_proposal.new_system)
+        self._old_to_hybrid_map = {}
+        self._new_to_hybrid_map = {}
+        self._hybrid_system_forces = dict()
+        self._old_positions = current_positions
+        self._new_positions = new_positions
+
+        self.softcore_alpha=0.5
+        self.softcore_beta=12*unit.angstrom**2
+
+        #prepare dicts of forces, which will be useful later
+        self._old_system_forces = {type(force).__name__ : force for force in self._old_system.getForces()}
+        self._new_system_forces = {type(force).__name__ : force for force in self._new_system.getForces()}
+
+        #get and store the nonbonded method from the system:
+        self._nonbonded_method = self._old_system_forces['NonbondedForce'].getNonbondedMethod()
+
+        #start by creating an empty system. This will become the hybrid system.
+        self._hybrid_system = openmm.System()
+
+        #begin by copying all particles in the old system to the hybrid system. Note that this does not copy the
+        #interactions. It does, however, copy the particle masses. In general, hybrid index and old index should be
+        #the same.
+        for particle_idx in range(self._topology_proposal.n_atoms_old):
+            particle_mass = self._old_system.getParticleMass(particle_idx)
+            hybrid_idx = self._hybrid_system.addParticle(particle_mass)
+            self._old_to_hybrid_map[particle_idx] = hybrid_idx
+
+            #If the particle index in question is mapped, make sure to add it to the new to hybrid map as well.
+            if particle_idx in self._topology_proposal.old_to_new_atom_map.keys():
+                particle_index_in_new_system = self._topology_proposal.old_to_new_atom_map[particle_idx]
+                self._new_to_hybrid_map[particle_index_in_new_system] = hybrid_idx
+
+        #Next, add the remaining unique atoms from the new system to the hybrid system and map accordingly.
+        #As before, this does not copy interactions, only particle indices and masses.
+        for particle_idx in self._topology_proposal.unique_new_atoms:
+            particle_mass = self._new_system.getParticleMass(particle_idx)
+            hybrid_idx = self._hybrid_system.addParticle(particle_mass)
+            self._new_to_hybrid_map[particle_idx] = hybrid_idx
+
```

Comment:
Should we copy virtual sites that are part of the environment though? I'd imagine that we should to support things like TIP4P?